### ISSUE TITLE
failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected

### ISSUE THREAD

**Issue by Thanushan1997** [Qengineering/Jetson-Nano-image#23]
When I try to list the physical GPU using the Tensorflow using the command "tf.config.list_physical_devices(device_type='GPU')". it gives me the following error . 

I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set
 I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcuda.so.1
 E tensorflow/stream_executor/cuda/cuda_driver.cc:328] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is 
 I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (nano): /proc/driver/nvidia/version does not exist

**Comment by Qengineering** [comment]
Does [this link](https://github.com/tensorflow/tensorflow/issues/16860) help you? It seems a common problem.

**Comment by Thanushan1997** [comment]
And while on the booting time i noticed, it's stuck on the below sectors. But I am able to login through serial console.

1. **“[ERROR] failed to start nvpmodel service”**
2. ** “[ERROR] Failed to start Docker Application Container Engine”**

the screenshots i have attached through this link : [https://drive.google.com/file/d/1MFFZYfXhKPIE5tj3f-U_0fqK0X4h5b4u/view?usp=share_link](url)

**Comment by Qengineering** [comment]
I'm afraid you have a corrupted image. 
Tested with a fresh download and a clean SD card I get:
![Screenshot from 2022-12-15 10-31-21](https://user-images.githubusercontent.com/44409029/207824547-aeb6f263-6f8c-479b-895d-b911667e4ac6.png)

**Comment by Thanushan1997** [comment]
I have downloaded the image from your github link. Is there any other link to download image?

**Comment by Qengineering** [comment]
The GitHub link must be OK. I will download the image from the Gdrive and install it once again, in case I missed something.
Here is my system info.
![Screenshot from 2022-12-15 13-19-03](https://user-images.githubusercontent.com/44409029/207858256-efafea8c-8507-4716-a466-e72f75ca17b2.png)

**Comment by Thanushan1997** [comment]
Hi
1) Jdk OS successfully booting the Jetson nano through USB slot. (clone the OS from jetson internal memory to SD card)
**Defaultly the board booted from internal storage, and then we Configured the jetson nano to boot from SD card slot**

below image will showed the details of some service files output : 
![workingjetson nano details](https://user-images.githubusercontent.com/61465914/208077913-519aa161-d3d5-4bea-bb4d-3820be6ab488.png)

2) But while flash your downloaded image, Its still stuck on the above error screen. Do you have found any luck on this matter?

**Comment by Qengineering** [comment]
I can confirm the image on GitHub has no nvpmodel issues. Downloaded for the second time and still no reproduction of your problem.
However, I see you have core version 32.7.3, while I still have version 32.6.1, even after `sudo apt-get update && sudo apt-get upgrade`. Perhaps this can be the cause?

**Comment by Thanushan1997** [comment]
Above picture from different image. not from downloaded this Repository.

**Comment by Qengineering** [comment]
In that case, I really don't know what causes your issue. I'm very sorry not able to help you.

**Comment by Thanushan1997** [comment]
Yeah i can understand. I just mentioned for your detailed references.

**Comment by Thanushan1997** [comment]
![image](https://user-images.githubusercontent.com/61465914/208638292-5bb3da31-0cec-49fc-b3d2-8d79c7bdaf66.png)


Our vendor said that in a default fresh jetson nano board, it doesn't boot the os from the SD card. If we flash your image in the SD card,  How can we configure the board to load the OS from the SD card, you haven't mentioned that in the GitHub link?

**Comment by Qengineering** [comment]
Dear @Thanushan1997,

Every Jetson Nano boots default from SD. It doesn't matter if you are using jetpack 4.5 or 4.6. If the file structure on the SD card is correct it will boot. Period. The only exception can be a dedicated Nano with altered hardware. 
If you can download [the original image](https://developer.nvidia.com/embedded/l4t/r32_release_v7.1/jp_4.6.1_b110_sd_card/jeston_nano/jetson-nano-jp461-sd-card-image.zip), flash it on a SD-card and get the Nano booted, it will also work with this image.
I've tested this image, with success, on the original Jetson Nano 4 GB board many times. 
I know it's a showstopper, but keep in mind that this image is being downloaded over 100+ per week, and we have never had any of the issues you face.
